the-collab-lab/tcl-12-smart-shopping-list
# Create unique token and save to localStorage then redirect to List view

## Description
This PR generates a new unique token for each user and saves it to localStorage, then redirects the user to List view.

- Created a button that onClick uses the getToken function to generate a new token
- Generated token is saved to localStorage using the setItem storage method.
- Once token is generated, the user is redirected to List view and the Navbar becomes visible
- If a user is on List view or Add an Item view with no token, they will be redirected to root directory to get a token
- If user has token, they will be redirected from root to list or add-item view automatically based on their preference



## Related Issue
Closes #3 



## Acceptance Criteria

- For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list
- Clicking the button/link generates a new token and saves it to localStorage
- Once the token has been created and saved, the user is redirected to the "list" view



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before
![Screen Shot 2020-10-14 at 12 59 01 AM](https://user-images.githubusercontent.com/24923742/95945728-b8c11980-0db0-11eb-9db3-a8112080df73.png)



### After
![get-new-token](https://user-images.githubusercontent.com/24923742/95946453-2d488800-0db2-11eb-9afb-08a4d3090551.gif)




## Testing Steps / QA Criteria



- From your terminal, pull down this branch with git pull origin gj-al-track-purchased-items and check that branch out with git checkout gj-al-track-purchased-items
- Click the "New Token" button to be redirected to List View
- Test paths by deleting your token in localStorage in the Inspector
     - Chrome: Inspect > Application > Local Storage > Right click to delete token
     - Safari: Inspect > Storage > Local Storage
     - Firefox: Inspect > Storage > Local Storage
